client/src/api/challenge.js

```diff
@@ -0,0 +1,18 @@
+export async function getAllTasks() {
+  const response = await fetch("/api/challenge");
+  const results = await response.json();
+  return results;
+}
+
+export async function countAllTasks() {
+  const response = await fetch("/api/challenge");
+  const amount = await response.json();
+  const length = amount.length;
+  return length;
+}
+
+export async function countDoneTasks() {
+  const response = await fetch("/api/challenge/done=false");
+  const amount = await response.json();
+  return amount;
+}
```

Comment:
its better to change the api return value, if you just need this specific information, you don't have to transfer all the data every time. just response an object instead of two independent ones

```suggestion
export async function amountTasks() {
  const response = await fetch("/api/challenge/statistic=true");
  return await response.json();
}
```